src/Symfony/Component/Security/Core/Exception/AccessDeniedException.php

```diff
@@ -18,8 +18,43 @@
  */
 class AccessDeniedException extends \RuntimeException
 {
+    private $attributes = [];
+    private $object;
+
     public function __construct($message = 'Access Denied.', \Exception $previous = null)
     {
         parent::__construct($message, 403, $previous);
     }
+
+    /**
+     * @return array
+     */
+    public function getAttributes()
+    {
+        return $this->attributes;
+    }
+
+    /**
+     * @param array $attributes
+     */
+    public function setAttributes(array $attributes)
+    {
+        $this->attributes = $attributes;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getObject()
+    {
+        return $this->object;
+    }
+
+    /**
+     * @param mixed $object
+     */
+    public function setObject($object)
```

Comment:
Though you are correct here, it's not always an object, it can be pretty much anything. I'm personally against voting on scalars or arrays, but it is allowed.